category_encoders/cat_boost.py

```diff
@@ -295,19 +165,21 @@ def _transform(self, X_in, y, mapping=None):
 
         return X
 
-    def get_feature_names(self):
-        """
-        Returns the names of all transformed / added columns.
+    def _more_tags(self):
```

Comment:
This was also introduced for "dynamic docstrings" and kept since it introduces useful information (also here you could imagine a test, also testing that the encoders where transform does not depend on y actually give the same result with and without y).  
I agree that `transform_depends_on_y` is a more suitable name